pkg/sql/catalog/systemschema/system.go

```diff
@@ -4188,6 +4265,128 @@ var (
 			}),
 	)
 
+	// SystemJobProgressTable is described in comment on JobProgressTableSchema.
+	SystemJobProgressTable = makeSystemTable(
+		JobProgressTableSchema,
+		systemTable(
+			catconstants.JobsProgressTableName,
+			descpb.InvalidID, // dynamically assigned
+			[]descpb.ColumnDescriptor{
+				{Name: "job_id", ID: 1, Type: types.Int},
+				{Name: "written", ID: 2, Type: types.TimestampTZ, DefaultExpr: &nowTZString},
+				{Name: "fraction", ID: 3, Type: types.Float, Nullable: true},
+				{Name: "resolved", ID: 4, Type: types.Decimal, Nullable: true},
+			},
+			[]descpb.ColumnFamilyDescriptor{
+				{
+					Name:            "primary",
+					ID:              0,
+					ColumnNames:     []string{"job_id", "written", "fraction", "resolved"},
+					ColumnIDs:       []descpb.ColumnID{1, 2, 3, 4},
+					DefaultColumnID: 4,
+				},
+			},
+			descpb.IndexDescriptor{
+				Name:                "primary",
+				ID:                  1,
+				Unique:              true,
+				KeyColumnNames:      []string{"job_id", "written"},
+				KeyColumnDirections: []catenumpb.IndexColumn_Direction{catenumpb.IndexColumn_ASC, catenumpb.IndexColumn_DESC},
+				KeyColumnIDs:        []descpb.ColumnID{1, 2},
+			}),
+	)
+
+	// SystemJobProgressHistoryTable is described in comment on JobProgressHistoryTableSchema.
+	SystemJobProgressHistoryTable = makeSystemTable(
+		JobProgressHistoryTableSchema,
+		systemTable(
+			catconstants.JobsProgressHistoryTableName,
+			descpb.InvalidID, // dynamically assigned
+			[]descpb.ColumnDescriptor{
+				{Name: "job_id", ID: 1, Type: types.Int},
+				{Name: "written", ID: 2, Type: types.TimestampTZ, DefaultExpr: &nowTZString},
+				{Name: "fraction", ID: 3, Type: types.Float, Nullable: true},
+				{Name: "resolved", ID: 4, Type: types.Decimal, Nullable: true},
+			},
+			[]descpb.ColumnFamilyDescriptor{
+				{
+					Name:            "primary",
+					ID:              0,
+					ColumnNames:     []string{"job_id", "written", "fraction", "resolved"},
+					ColumnIDs:       []descpb.ColumnID{1, 2, 3, 4},
+					DefaultColumnID: 4,
+				},
+			},
+			descpb.IndexDescriptor{
+				Name:                "primary",
+				ID:                  1,
+				Unique:              true,
+				KeyColumnNames:      []string{"job_id", "written"},
+				KeyColumnDirections: []catenumpb.IndexColumn_Direction{catenumpb.IndexColumn_ASC, catenumpb.IndexColumn_DESC},
+				KeyColumnIDs:        []descpb.ColumnID{1, 2},
+			}),
+	)
+
+	// SystemJobStatusTable is described in comment on JobStatusTableSchema.
+	SystemJobStatusTable = makeSystemTable(
+		JobStatusTableSchema,
+		systemTable(
+			catconstants.JobsStatusTableName,
+			descpb.InvalidID, // dynamically assigned
+			[]descpb.ColumnDescriptor{
+				{Name: "job_id", ID: 1, Type: types.Int},
+				{Name: "written", ID: 2, Type: types.TimestampTZ, DefaultExpr: &nowTZString},
+				{Name: "message", ID: 3, Type: types.String, Nullable: true},
```

Comment:
running_status can be absent so I think we thought this could be too. But I suppose now that we're keeping it to one row you could just delete it rather than writing a new one with NULL.

Reply:
made it not-null